ddsp/training/docker/user_interaction.py

```diff
@@ -1,84 +1,235 @@
-                    '/<IMAGE_REPO_NAME>:<IMAGE_TAG> :\n')
+  if FLAGS.project_id:
+    project_id = FLAGS.project_id
+  else:
+    project_id = subprocess.getoutput('gcloud config get-value project')
+    if project_id == '(unset)':
+      project_id = input(prompt('value', 'your project ID'))
+    check_project_id = 'gcloud projects describe ' + project_id
+    while not subprocess.getoutput(check_project_id).startswith('createTime'):
+      print('Project ID not valid.')
+      project_id = input(prompt('value', 'your project ID'))
+      check_project_id = 'gcloud projects describe ' + project_id
+
+  image_uri = 'gcr.io/' + project_id + '/ddsp_training:train_job'
   input_dict['image_uri'] = image_uri
 
-  job_name = input('\nInsert a job name:\n')
+  job_name = 'training_job_' + str(int((datetime.datetime.now()\
+             - datetime.datetime(1970,1,1)).total_seconds()))
   input_dict['job_name'] = job_name
 
-  region = input('\nInsert the region you want to train '
-                  'your model in or skip for the default '
-                  'value (europe-west4): ')
-  if region == '':
-    region = 'europe-west4'
+  if FLAGS.region:
+    region = FLAGS.region
+  else:
+    region = subprocess.getoutput('gcloud config get-value compute/region')
+    if region == '(unset)':
+      region = input('\nInsert the region you want to train '
+                      'your model in or skip for the default '
+                      'value (europe-west4): ')
+      if region == '':
+        region = 'europe-west4'
   input_dict['region'] = region
 
-  batch_size = input('\nInsert batch size '
-                      'or skip for the default value (128): ')
-  if batch_size == '':
-    batch_size = '128'
+  if FLAGS.batch_size:
+    batch_size = FLAGS.batch_size
+  else:
+    batch_size = input(prompt('value', 'batch size', skip=True, arg='128'))
+    if batch_size == '':
+      batch_size = '128'
   input_dict['batch_size'] = batch_size
 
-  learning_rate = input('\nInsert the learning rate '
-                        'or skip for the default value (0.001): ')
-  if learning_rate == '':
-    learning_rate = '0.001'
+  if FLAGS.learning_rate:
+    learning_rate = FLAGS.learning_rate
+  else:
+    learning_rate = input(prompt('value', 'the learning rate',\
+                                 skip=True, arg='0.001'))
+    if learning_rate == '':
+      learning_rate = '0.001'
   input_dict['learning_rate'] = learning_rate
 
-  no_of_steps = input('\nInsert the number of steps for training '
-                      'or skip for the default value (15000): ')
-  if no_of_steps == '':
-    no_of_steps = '15000'
-  input_dict['no_of_steps'] = no_of_steps
+  if FLAGS.num_steps:
+    num_steps = FLAGS.num_steps
+  else:
+    num_steps = input(prompt('value', 'the number of steps for training',\
+                             skip=True, arg='15000'))
+    if num_steps == '':
+      num_steps = '15000'
+  input_dict['num_steps'] = num_steps
```

Comment:
also, since you aren't using num_steps elsewhere you can write directly to input_dict from get_value() and choose a shorter name for `input_dict`, like `args`